Application/EdFi.Ods.Api/Extensions/ApplicationBuilderExtensions.cs

```diff
@@ -21,7 +23,26 @@ public static IApplicationBuilder UseOpenApiMetadata(this IApplicationBuilder bu
         public static IApplicationBuilder UseEdFiApiAuthentication(this IApplicationBuilder builder)
             => builder.UseMiddleware<EdFiApiAuthenticationMiddleware>();
 
-        public static IApplicationBuilder UseInstanceYearSpecific(this IApplicationBuilder builder)
-            => builder.UseMiddleware<InstanceYearSpecificMiddleware>();
+        public static IApplicationBuilder UseSchoolYearRouteContext(this IApplicationBuilder builder, ApiSettings apiSettings)
```

Comment:
I'm not a fan of this pattern.  The convention of "UseXXX" with builders is that you expect it to do something, but you're hiding this with api mode checks inside, so in some cases it may do nothing.  The Api mode checks should be handled during startup prior to calling this Use method, and the intent of this Use method should be to actually apply the middleware.

Reply:
ok will revert